src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -454,6 +459,39 @@ protected override int AddVariable(VariableIdentifier identifier)
             return _variables.Add(identifier);
         }
 
+        private void AddUnvisitedPlaceholder(BoundValuePlaceholderBase placeholder, BoundExpression unvisited)
```

Comment:
>AddUnvisitedPlaceholder [](http://example.com/codeflow?start=21&length=23)

All the names sound to me like we are dealing with unvisited placeholders. However, it feels to me that the "unvisited" part is about the other expression node that, in general, is not a placeholder. I would probably simply dropped the "unvisited" from the names, or made them clearer, if I thought it is important to reflect the fact in names. For example, ```_placeholderToUnvisitedExpressionOpt``` instead of ```_unvisitedPlaceholdersOpt```, etc.